src/main/java/ch/njol/skript/expressions/ExprFormatTime.java

```diff
@@ -51,35 +59,50 @@
 		Skript.registerExpression(ExprFormatTime.class, String.class, ExpressionType.PROPERTY, "%dates% formatted [human-readable] [(with|as) %-string%]");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private SimpleDateFormat format;
 
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<? extends String> customFormat;
+
 	@Override
-	@SuppressWarnings("null")
+	@SuppressWarnings({"null", "unchecked"})
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr((Expression<? extends Date>) exprs[0]);
-		if (exprs[1] != null) {
-			if (!(exprs[1] instanceof Literal)) {
-				VariableString str = (VariableString) exprs[1];
-				if (!str.isSimple()) {
-					Skript.error("Date format must not contain variables!");
-					return false;
+		customFormat = (Expression<? extends String>) exprs[1];
+
+		if (customFormat != null) {
+			if (!(customFormat instanceof Literal) && customFormat instanceof VariableString) {
+				VariableString str = (VariableString) customFormat;
+				if (str.isSimple()) {
+					try {
+						format = new SimpleDateFormat(customFormat.getSingle(null));
+					} catch (Exception e) {
+						Skript.error("Incorrect date format used: " + e.getMessage());
+						return false;
+					}
 				}
 			}
-			format = new SimpleDateFormat((String) exprs[1].getSingle(null));
-		} else {
-			format = new SimpleDateFormat(defaultFormat);
 		}
-		
+
 		return true;
 	}
 
-
 	@Override
 	protected String[] get(Event e, Date[] source) {
 		return get(source, new Getter<String, Date>() {
 			@Override
 			public String get(Date date) {
+				if (customFormat != null) {
+					try {
+						format = new SimpleDateFormat(customFormat.getSingle(e));
+					} catch (Exception ex) {
+//						Skript.warning("Incorrect date format used: " + ex.getMessage());
+						format = new SimpleDateFormat(defaultFormat);
+					}
+				} else {
+					format = new SimpleDateFormat(defaultFormat);
```

Comment:
if this suggestion is implemented, this else case will no longer be needed https://github.com/SkriptLang/Skript/pull/4664/files#r827513338